File: 02-01-2025_Count_vowel_strings_in_range.java
private boolean isVowel(char ch) {
    return ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u';
}

private boolean search(String s) {
    boolean startsWithVowel = isVowel(s.charAt(0));
    boolean endsWithVowel = isVowel(s.charAt(s.length() - 1));
    return startsWithVowel && endsWithVowel;
}

public int[] vowelStrings(String[] words, int[][] queries) {
    int n = words.length;
    int[] binary = new int[n];
    
    for (int i = 0; i < n; i++) {
        if (search(words[i])) {
            binary[i] = 1;
        }
    }
    
    for (int i = 1; i < n; i++) {
        binary[i] += binary[i - 1];
    }
    
    int[] ans = new int[queries.length];
    for (int i = 0; i < queries.length; i++) {
        int l = queries[i][0];
        int r = queries[i][1];
        
        int temp;
        if (l == 0) {
            temp = binary[r];
        } else {
            temp = binary[r] - binary[l - 1];
        }
        ans[i] = temp;
    }
    
    return ans;
}

File: 02-01-2025_Count_vowel_strings_in_range.py
def is_vowel(self, ch):
    return ch in {'a', 'e', 'i', 'o', 'u'}

def search(self, s):
    starts_with_vowel = self.is_vowel(s[0])
    ends_with_vowel = self.is_vowel(s[-1])
    return starts_with_vowel and ends_with_vowel

def vowelStrings(self, words, queries):
    """
    :type words: List[str]
    :type queries: List[List[int]]
    :rtype: List[int]
    """
    n = len(words)
    binary = [0] * n
    
    for i in range(n):
        if self.search(words[i]):
            binary[i] = 1
    
    for i in range(1, n):
        binary[i] += binary[i - 1]
    
    ans = []
    for query in queries:
        l, r = query
        l -= 1  # Convert 1-based to 0-based index
        
        if l < 0:
            temp = binary[r]
        else:
            temp = binary[r] - binary[l]
        
        ans.append(temp)
    
    return ans
